### DISCUSSION TITLE
S3 Upload POST request returns 403 Forbidden while PUT request work.

### DISCUSSION THREAD

**Discussion by satriobio** [minio/minio#16091] [Q&A] [answered]
Hi Everyone,

I'm trying to deploy a Nuxt + Minio + Nginx web app using docker compose

I'm using quay.io/minio/minio:RELEASE.2022-10-29T06-21-33Z image and I have followed recommended nginx setup.
https://min.io/docs/minio/linux/integrations/setup-nginx-proxy-with-minio.html

```
...

location /mybucket/ {
   proxy_set_header X-Real-IP $remote_addr;
   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   proxy_set_header X-Forwarded-Proto $scheme;
   proxy_set_header Host $http_host;

   proxy_connect_timeout 300;
   proxy_http_version 1.1;
   proxy_set_header Connection "";
   chunked_transfer_encoding off;
   proxy_pass http://storage:9000;
 }

location / {
   proxy_set_header X-Real-IP $remote_addr;
   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   proxy_set_header X-Forwarded-Proto $scheme;
   proxy_set_header Host $http_host;
   proxy_connect_timeout 300;
   proxy_http_version 1.1;
   proxy_set_header Connection "";
   chunked_transfer_encoding off;

   proxy_pass http://frontend:3000;
 }
```

I can perform S3Client, ListObjectsCommand, PutObjectCommand, Upload function from aws-sdk v3 just fine in local machine. But when I deploy the app on my GCP VM (load balancer, and domain configured) only the Upload function not working.

It returns only 403 Forbidden specifically on the very last POST request.

I'm not sure the main cause of the issue, and i have tried to find the solutions for couple of days now. It seems no one have reported this before, I wonder if anyone familiar with this kind of issue.

Thank you.

**Comment by harshavardhana** [comment, marked as answer]
This is because you have a bucket referenced here and this is not allowed S3 API always stays at `location /` not at `location /mybucket` having that causes signature issues. 

You cannot have your LB configured in this manner. 

https://github.com/minio/minio/blob/master/docs/orchestration/docker-compose/nginx.conf look at this example on how to successfully configure nginx LB
